onnxruntime/core/providers/mkldnn/subgraph/mkldnn_custom_op.h

```diff
@@ -0,0 +1,283 @@
+        std::shared_ptr<MklPool<T>> kernel;
+        kernel.reset(new MklPool<T>(mklnode, params.provider, params.mkl_context));
+        kernel->ReadAttributes(params.attributes, os.str());
+        for (auto& index : mklnode.parent_nodes) {
+          kernel->parents_.push_back(context_.kernels[index]);
+        }
+        context_.kernels.push_back(kernel);
+      } else if (mklnode.name == "GlobalAveragePool") {
+        std::ostringstream os;
+        os << "GlobalAveragePool-" << mklnode.node_index << "-";
+        std::shared_ptr<MklPool<T>> kernel;
+        kernel.reset(new MklPool<T>(mklnode, params.provider, params.mkl_context));
+        kernel->ReadAttributes(params.attributes, os.str());
+        for (auto& index : mklnode.parent_nodes) {
+          kernel->parents_.push_back(context_.kernels[index]);
+        }
+        context_.kernels.push_back(kernel);
+      } else if (mklnode.name == "Sum") {
+        std::ostringstream os;
+        os << "Sum-" << mklnode.node_index << "-";
+        std::shared_ptr<MklSum<T>> kernel;
+        kernel.reset(new MklSum<T>(mklnode, params.provider, params.mkl_context));
+        kernel->ReadAttributes(params.attributes, os.str());
+        for (auto& index : mklnode.parent_nodes) {
+          kernel->parents_.push_back(context_.kernels[index]);
+        }
+        context_.kernels.push_back(kernel);
+      } else if (mklnode.name == "LRN") {
+        std::ostringstream os;
+        os << "LRN-" << mklnode.node_index << "-";
+        std::shared_ptr<MklLrn<T>> kernel;
+        kernel.reset(new MklLrn<T>(mklnode, params.provider, params.mkl_context));
+        kernel->ReadAttributes(params.attributes, os.str());
+        for (auto& index : mklnode.parent_nodes) {
+          kernel->parents_.push_back(context_.kernels[index]);
+        }
+        context_.kernels.push_back(kernel);
+      }
+    }
+  }
+
+  Status Compute(const ONNXRunTimeTensor* input_tensors, ONNXRunTimeTensor* const output_tensors) {
+    Status status;
+    for (auto& kernel : context_.kernels) {
+      status = kernel->Bind(input_tensors, output_tensors);
+      if (!status.IsOK())
+        break;
+    }
+    if (status.IsOK())
+      context_.stream->submit(context_.net);
+    return status;
+  }
+
+  ~SubgraphPrimitive() = default;
+
+ private:
+  struct SubgraphContext {
+    std::unique_ptr<mkldnn::stream> stream;
+    std::vector<mkldnn::primitive> net;
+    std::vector<std::shared_ptr<MklKernel>> kernels;
+
+    SubgraphContext() : stream(nullptr) {}
+  };
+
+  void Initialize(const ONNXRunTimeTensor* input_tensors) {
+    // Propagate mkldnn block format
+    // dst format of current node to src format of next node
+    mkldnn::memory::format source_format = mkldnn::memory::format::any;  // ONNXRuntime format
+    for (auto& kernel : context_.kernels) {
+      Status status = kernel->CreatePrimitives(input_tensors, cpu_engine_, context_.net, source_format);
+      if (status.IsOK())
```

Comment:
Looks like Initialize will fail silently. Can you make it return a status code and throw in the constructor if a failure is encountered?